### DISCUSSION TITLE
84889d4 : boards won't build

### DISCUSSION THREAD

**Discussion by scaprile** [zephyrproject-rtos/zephyr#77662] [General]
The following boards stopped building after https://github.com/zephyrproject-rtos/zephyr/commit/84889d478362876e3a2bf4669e205def14c6794b:

nucleo_h563zi
stm32h573i_dk
nucleo_h743zi
stm32h735g_disco
stm32h745i_disco/stm32h745xx/m7
stm32h747i_disco/stm32h747xx/m7
nucleo_h753zi
nucleo_h755zi_q/stm32h755xx/m7

The error is this one:
```
zephyr/drivers/ethernet/eth_stm32_hal.c:62:92: error: pasting ")" and "_ORD" does not give a valid preprocessing token
   62 |             DEVICE_DT_GET(DT_CHILD(DT_INST_CHILD(n, mdio), __CONCAT(ethernet_phy_, PHY_ADDR)))
```
looks like `CONFIG_ETH_STM32_HAL_PHY_ADDRESS` does not get a value (hence also `PHY_ADDR`)

**`nucleo_h563zi`, for example, builds again when rolling back that commit.**

My prj.conf is bare minimum
```
CONFIG_NETWORKING=y
CONFIG_NET_IPV4=y
CONFIG_NET_IPV6=y
CONFIG_NET_TCP=y
CONFIG_NET_UDP=y
CONFIG_NET_DHCPV4=y
CONFIG_NET_SOCKETS=y
CONFIG_NET_SOCKETS_POLL_MAX=32
CONFIG_POSIX_API=y
CONFIG_POSIX_MAX_FDS=32
CONFIG_NET_MAX_CONN=10
CONFIG_NET_MAX_CONTEXTS=10
CONFIG_NET_CONFIG_SETTINGS=y
CONFIG_NET_CONNECTION_MANAGER=y
CONFIG_NET_LOG=y

CONFIG_LOG=y
CONFIG_ISR_STACK_SIZE=2048
CONFIG_MAIN_STACK_SIZE=8192
CONFIG_IDLE_STACK_SIZE=1024

CONFIG_MINIMAL_LIBC=y
CONFIG_MINIMAL_LIBC_RAND=y
CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=32768
```
Is this an error on my side ? Something essential is missing that breaks Kconfig reading ?

These boards, nevertheless, build OK with that very same prj.conf:

nucleo_f207zg
nucleo_f429zi
nucleo_f746zg
nucleo_f767zi

**Comment by jukkar** [comment]
Hi @scaprile please create an issue for this as then relevant people can investigate it better.

**Comment by jukkar** [comment]
There is an issue and also a fix so closing this.
